core/meterer/accountledger.go

```diff
@@ -0,0 +1,364 @@
+package meterer
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/core"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// AccountLedger defines the standard interface for tracking account payment state,
+// including both on-chain settings and local usage tracking. This interface abstracts
+// the differences between client-side in-memory tracking and server-side persistent storage.
+//
+// Implementation Patterns:
+//   - LocalAccountLedger: In-memory tracking for clients
+//   - Future: DatabaseAccountLedger for persistent server-side tracking
+//   - Future: DistributedAccountLedger for consensus-based tracking
+type AccountLedger interface {
+	// RecordReservationUsage attempts to record symbol usage against the ledger's stored
+	// reservations for the specified quorums. It atomically checks usage against bin limits,
+	// handles overflows, and updates the underlying storage.
+	//
+	// The method performs the following operations atomically:
+	//   1. Validates that reservations exist and are currently active for all quorums
+	//   2. Calculates the time period bin for the given timestamp
+	//   3. Checks if the requested symbols fit within rate limits for each quorum
+	//   4. Handles overflow to future periods if current period is full
+	//   5. Updates all period records or rolls back entirely on any failure
+	//
+	// Parameters:
+	//   ctx: Context for cancellation and timeouts (currently unused in LocalAccountLedger)
+	//   accountID: The account identifier (used for logging and future database implementations)
+	//   timestampNs: The timestamp in nanoseconds when the usage occurred
+	//   numSymbols: The number of symbols to be charged against reservations
+	//   quorumNumbers: The list of quorums that will be charged for this usage
+	//   params: Payment vault parameters containing quorum configurations
+	//
+	// Returns:
+	//   nil if the usage was successfully recorded
+	//   error if the usage cannot be accommodated:
+	//     - "reservation not found for quorum X": No reservation exists for quorum
+	//     - "reservation limit exceeded": Usage would exceed rate limits
+	//     - "reservation expired": Current time is outside reservation window
+	//     - "quorum config not found for X": Missing protocol configuration
+	RecordReservationUsage(
```

Comment:
Is there a reason that `RecordReservationUsage` and `RecordOnDemandUsage` should be separate methods on the interface?

It seems that a single method accepting details about a payment would be sufficient- I don't see why you would ever handle one and not the other

Reply:
I see, okay. do you mind reviewing the debit slip PR first, and I'll address these comments in my tmr morning?